src/SerilogTracing/Instrumentation/ActivityInstrumentation.cs

```diff
@@ -248,6 +248,50 @@ class TextException(
         public override string ToString() => toString ?? "No information available.";
     }
 
+    /// <summary>
+    /// Suppress the current activity, linking any newly created activities with its parent.
+    /// </summary>
+    /// <returns>A value that will restore the suppressed activity on dispose. If there is no current activity, it
+    /// will return null.</returns>
+    /// <remarks>
+    /// This method is intended to suppress activities from outside sources when you want to replace them with your own.
+    /// 
+    /// It takes <see cref="Activity.Current" />, unsets the <see cref="ActivityTraceFlags.Recorded" /> flag,
+    /// sets <see cref="Activity.IsAllDataRequested" /> to false, and re-assigns <see cref="Activity.Current" /> to
+    /// its <see cref="Activity.Parent" />.
+    ///
+    /// When the returned <see cref="IDisposable" /> is disposed, the <see cref="Activity.Current" /> will be restored,
+    /// but the restored activity will remain unrecorded.
+    /// </remarks>
+    public static IDisposable? SuppressCurrentActivity()
```

Comment:
It seems like there are a few different suppression scenarios:

 * Make current activity unrecorded
 * Replace current activity with `null`
 * Replace current activity with `null` and make current activity unrecorded
 * Replace current activity with its parent
 * Replace current activity with its parent and make the parent unrecorded

And possibly some more variations..? The API naming sounds like the feature is general, but it picks a specific scenario. Perhaps this needs a rethink?

Reply:
Happy to give this some more thought. The naming is general, but also vague, so without the docs it’s not clear what it really does. The most obvious path from here to supporting other scenarios is via overloads that accept various flags, but I don’t think that’s an ideal place to land.

The method is an encapsulated strategy for suppressing the current activity, which I think is important, otherwise you can easily do it wrong.

This approach is “link new activities to the current parent, ignoring the current activity”. Maybe we can find a name around replacing an activity with its parent, like `InParentActivityContext` and add a single flag for suppressing the current activity too?

Reply:
Something like that could work. There's also the case to consider where the current activity has a remote parent, so `Parent` will be null and although `Current` appears to have a parent (non-default `ParentSpanId`), that parent won't become active when `Current` is suppressed.

Reply:
Hmm, perhaps we are better off untangling these concepts. So a method like `ActivityInstrumentation.SuppressActivity(Activity activity)` which does the flags on the activity, and another to `SwapCurrentActivity(Activity? replacement)` which sets current to `replacement` and restores it on dispose?